includes/admin/abstract/class-rop-services-abstract.php

```diff
@@ -904,4 +904,54 @@ function( $d ) {
 		return $valid;
 	}
 
+	/**
+	 * Capture logs on ROP API.
+	 *
+	 * @since   9.1.3
+	 * @access  public
+	 *
+	 * @return bool
+	 */
+	public function save_logs_on_rop( $data = array() ) {
+		if ( empty( $data ) ) {
+			return false;
+		}
+
+		$license_key    = apply_filters( 'product_rop_license_key', '' );
+		$license_status = apply_filters( 'product_rop_license_status', 'invalid' );
+
+		// Send API request.
+		$response = wp_remote_post(
+			ROP_POST_LOGS_API,
+			apply_filters(
+				'rop_post_capture_logs_api_args',
+				array(
+					'timeout' => 100,
+					'body'    => array_merge(
+						array(
+							'website'        => get_site_url(),
+							'license'        => $license_key,
+							'license_status' => $license_status,
+						),
+						$data
+					),
+				)
+			)
+		);
+
+		$this->logger->info( '[ROP] Response: ' . print_r( $response, true ) );
+
+		if ( ! is_wp_error( $response ) ) {
+			$body          = json_decode( wp_remote_retrieve_body( $response ) );
+			$response_code = wp_remote_retrieve_response_code( $response );
+
+			$this->logger->info( '[ROP] Response Body: ' . wp_json_encode( $body ) );
+
+			if ( 200 === $response_code ) {
+				return $body->status;
+			}
+		}
+		return false;
```

Comment:
you can simplify to:

```php
if ( is_wp_error( $response ) ) {
	return false;
}

$body          = json_decode( wp_remote_retrieve_body( $response ) );
$response_code = wp_remote_retrieve_response_code( $response );

$this->logger->info( '[ROP] Response Body: ' . wp_json_encode( $body ) );
return 200 === $response_code ? $body->status : false;
```

If you apply the suggestion from https://github.com/Codeinwp/tweet-old-post/pull/1025/files#r1935139755, you can drop the last return.

Reply:
@Soare-Robert-Daniel I've implemented the suggested changes. Please review them in the latest commit.